lib/spack/spack/build_systems/rocm.py

```diff
@@ -145,7 +145,11 @@ class ROCmPackage(PackageBase):
     depends_on("hip +rocm", when="+rocm")
 
     # need amd gpu type for rocm builds
+    compilers_supporting_rocm = ("cce", "rocmcc")
     conflicts("amdgpu_target=none", when="+rocm")
+    for compiler in spack.compilers.supported_compilers():
+        if compiler not in compilers_supporting_rocm:
+            conflicts(f"%{compiler}", when="+rocm")
```

Comment:
This breaks existing workflows that don't use `CachedCMakePackage` and use `%gcc` as their base and rely on the current `hipcc` workaround in many spackages.

Putting `%rocmcc` has been usually a last resort and always caused issues in the past. By default it would also propagate down to dependencies which can cause problems (e.g. packages also need a gfortran compiler).

Reply:
Note, I'm all for removing the `hipcc` part, since that would allow me to also start using CachedCMakePackage for https://github.com/spack/spack/pull/43517. I only warn about adding the conflict themselves, since they may have unintended consequences for people who started relying on the "magic" and are reluctant to wrestle with the concretizer when it gets convinced that `rocmcc` must go somewhere else too.

Reply:
@rbberger I removed the `hipcc` override in this merged PR #43644.

This PR is slightly out of date and there were a lot of discussions about related topics.